backend/src/schema/resolvers/reports.js

```diff
@@ -82,24 +84,24 @@ export default {
       const limit = params.first && typeof params.first === 'number' ? `LIMIT ${params.first}` : ''
 
       const reportReadTxPromise = session.readTransaction(async tx => {
-        const allReportsTransactionResponse = await tx.run(
-          `
-            MATCH (report:Report)-[:BELONGS_TO]->(resource)
-            WHERE (resource:User OR resource:Post OR resource:Comment)
-            ${filterClause}
-            WITH report, resource,
-            [(submitter:User)-[filed:FILED]->(report) |  filed {.*, submitter: properties(submitter)} ] as filed,
-            [(moderator:User)-[reviewed:REVIEWED]->(report) |  reviewed {.*, moderator: properties(moderator)} ] as reviewed,
-            [(resource)<-[:WROTE]-(author:User) | author {.*} ] as optionalAuthors,
-            [(resource)-[:COMMENTS]->(post:Post) | post {.*} ] as optionalCommentedPosts,
-            resource {.*, __typename: labels(resource)[0] } as resourceWithType
-            WITH report, optionalAuthors, optionalCommentedPosts, reviewed, filed,
-            resourceWithType {.*, post: optionalCommentedPosts[0], author: optionalAuthors[0] } as finalResource
-            RETURN report {.*, resource: finalResource, filed: filed, reviewed: reviewed }
-            ${orderByClause}
-            ${offset} ${limit}
-          `,
-        )
+        const cypher = `
+          MATCH (report:Report)-[:BELONGS_TO]->(resource)
+          WHERE (resource:User OR resource:Post OR resource:Comment)
+          ${filterClause}
+          WITH report, resource,
+          [(submitter:User)-[filed:FILED]->(report) |  filed {.*, submitter: properties(submitter)} ] as filed,
+          [(moderator:User)-[reviewed:REVIEWED]->(report) |  reviewed {.*, moderator: properties(moderator)} ] as reviewed,
+          [(resource)<-[:WROTE]-(author:User) | author {.*} ] as optionalAuthors,
+          [(resource)-[:COMMENTS]->(post:Post) | post {.*} ] as optionalCommentedPosts,
+          resource {.*, __typename: labels(resource)[0] } as resourceWithType
+          WITH report, optionalAuthors, optionalCommentedPosts, reviewed, filed,
+          resourceWithType {.*, post: optionalCommentedPosts[0], author: optionalAuthors[0] } as finalResource
+          RETURN report {.*, resource: finalResource, filed: filed, reviewed: reviewed }
+          ${orderByClause}
+          ${offset} ${limit}
+        `
+        const allReportsTransactionResponse = await tx.run(cypher)
+        log(allReportsTransactionResponse)
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-10T10:16:11Z" title="Tuesday, December 10th 2019, 11:16:11 am +01:00">Dec 10, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -82,24 +84,24 @@ export default {
       const limit = params.first && typeof params.first === 'number' ? `LIMIT ${params.first}` : ''
 
       const reportReadTxPromise = session.readTransaction(async tx => {
-        const allReportsTransactionResponse = await tx.run(
-          `
-            MATCH (report:Report)-[:BELONGS_TO]->(resource)
-            WHERE (resource:User OR resource:Post OR resource:Comment)
-            ${filterClause}
-            WITH report, resource,
-            [(submitter:User)-[filed:FILED]->(report) |  filed {.*, submitter: properties(submitter)} ] as filed,
-            [(moderator:User)-[reviewed:REVIEWED]->(report) |  reviewed {.*, moderator: properties(moderator)} ] as reviewed,
-            [(resource)<-[:WROTE]-(author:User) | author {.*} ] as optionalAuthors,
-            [(resource)-[:COMMENTS]->(post:Post) | post {.*} ] as optionalCommentedPosts,
-            resource {.*, __typename: labels(resource)[0] } as resourceWithType
-            WITH report, optionalAuthors, optionalCommentedPosts, reviewed, filed,
-            resourceWithType {.*, post: optionalCommentedPosts[0], author: optionalAuthors[0] } as finalResource
-            RETURN report {.*, resource: finalResource, filed: filed, reviewed: reviewed }
-            ${orderByClause}
-            ${offset} ${limit}
-          `,
-        )
+        const cypher = `
+          MATCH (report:Report)-[:BELONGS_TO]->(resource)
+          WHERE (resource:User OR resource:Post OR resource:Comment)
+          ${filterClause}
+          WITH report, resource,
+          [(submitter:User)-[filed:FILED]->(report) |  filed {.*, submitter: properties(submitter)} ] as filed,
+          [(moderator:User)-[reviewed:REVIEWED]->(report) |  reviewed {.*, moderator: properties(moderator)} ] as reviewed,
+          [(resource)<-[:WROTE]-(author:User) | author {.*} ] as optionalAuthors,
+          [(resource)-[:COMMENTS]->(post:Post) | post {.*} ] as optionalCommentedPosts,
+          resource {.*, __typename: labels(resource)[0] } as resourceWithType
+          WITH report, optionalAuthors, optionalCommentedPosts, reviewed, filed,
+          resourceWithType {.*, post: optionalCommentedPosts[0], author: optionalAuthors[0] } as finalResource
+          RETURN report {.*, resource: finalResource, filed: filed, reviewed: reviewed }
+          ${orderByClause}
+          ${offset} ${limit}
+        `
+        const allReportsTransactionResponse = await tx.run(cypher)
+        log(allReportsTransactionResponse)
```


I don't know why this change was made(?) 
I know that there are some who like to save the `cypher` statement to a variable, but I don't see any actual benefit of doing it. Am I missing something?
It seems to be a matter of code style,  I prefer to use the `cypher` statement directly in the `tx.run`, others prefer this style.
Is there some reason you made this change that is tangible, or just a preference?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-12-10T16:15:46Z" title="Tuesday, December 10th 2019, 5:15:46 pm +01:00">Dec 10, 2019</time>_

---


Nah, it's a leftover of a previous commit. I wanted to log `cypher` and `params`. Then I discovered that the `allReportsTransactionResponse.summary` has both arguments as well. So just create a logging function for the response instead of loggin the two variables.

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-10T16:17:22Z" title="Tuesday, December 10th 2019, 5:17:22 pm +01:00">Dec 10, 2019</time>_

---


ok, this `PR` has been including in the other https://github.com/Human-Connection/Human-Connection/pull/2470, correct?
If so, can we close this?